### DISCUSSION TITLE
Should `sample_type` be required?

### DISCUSSION THREAD

**Discussion by bmeluch** [microbiomedata/nmdc-schema#2141] [Ideas]
We have a slot for [`sample_type`](https://microbiomedata.github.io/berkeley-schema-fy24/sample_type/), but it is "recommended" not "required" and is not consistently used. Its range is `SampleTypeEnum` which includes values similar to the options in the submission portal: soil, soil - water extract, plant associated, sediment, water

There have been instances recently where it would be really nice to know the type of a sample. (e.g. [checking that depth values meet our standards by sample type](https://github.com/microbiomedata/issues/issues/764), [in process notebook to visualize NOM data](https://github.com/microbiomedata/nmdc_notebooks/tree/34-create-r-notebook-to-explore-processed-data-for-a-second-omics-data-type) )

The sample type can be derived from required slots that have information for all biosamples (yay!) 
...but those slots are the environmental triad, so you have to comb through ENVO terms/IDs, which is really not user-friendly (boo)

I know there was reasoning for not having `sample_type` required, but I don't know what it is. Thoughts?

**Comment by bmeluch** [comment]
@mslarae13 as discussed last Thursday I think

**Comment by mslarae13** [comment]
Related to 
- https://github.com/microbiomedata/issues/discussions/766

Capturing sample type would be required for success.

**Comment by kheal** [comment]
If we _can_ derive `samp_type` from the ENVO terms/IDs, I think it would be better if this population was done without human interference.  

In other words, can we populate `samp_type` automatically from ENVO terms/IDs rather than requiring it upon submission (and therefore opening up the opportunity for error by mismatching the `samp_type` with ENVO terms/IDs)?  I'm not sure how this could be implemented, but that is how I would approach it - make a set of rules or functions to populate it from already required fields.

**Reply by mslarae13** [reply]
>If we can derive samp_type from the ENVO terms/IDs, 

That's too risky IMO. What if we're wrong? It should come form the person... and the environmental triad terms are not peoples strong suit.
But for what's currently in Mongo, maybe we have to?

> I think it would be better if this population was done without human interference.

I don't think we need additional human interference in the submission portal, we populate it with the extension they select.
The problem is data in mongo needs migrated to include that attribute, or it's invalid.

**Comment by bmeluch** [comment]
> The problem is data in mongo needs migrated to include that attribute, or it's invalid.

@turbomam and I have both done sample type categorization by ENVO term recently, we can compare notes, this is doable

my very manual R code using only env_medium values currently in mongo: https://github.com/microbiomedata/issues/issues/612#issuecomment-2239725442

(Mark do you have a link to the ENVO based sample typing you showed in the 1pm meeting yesterday? I can't find it)

**Reply by mslarae13** [reply]
It was a google sheet

**Reply by turbomam** [reply]
Could this be what you are talking about?

[nmdc-production-biosamples-env_package-predictions](https://docs.google.com/spreadsheets/d/1qlFvuy3Jw2u5Z-8txc6z2woHIZiaatneMSrjIFOWjRU/edit?gid=1679051587#gid=1679051587)

I suggest using the contents of the `predicted_curated_env_package` column to fill in each Biosample's `env_package`. I ahven't really thought about actions related to `sample_type`, but I will now.

**Comment by bmeluch** [comment]
more thoughts... once we get the environmental package auto importing from the submission portal, does it go into `env_package` and/or `sample_type`? 

`sample_type` says "This can vary from 'environmental package' if the sample is an extraction." which I read as "these two slots are the same unless the sample is an extraction" 

I know `env_package` comes from MIxS. It seems like our [`SampleTypeEnum`](https://microbiomedata.github.io/nmdc-schema/SampleTypeEnum/) currently captures some MIxS extensions, and our [`env_package`](https://microbiomedata.github.io/nmdc-schema/env_package/) is currently just a string. Seems like they should both be enums maybe?

I'm not fussed about which slot ends up being required. I just think at least one of them should be, so that it's easy to know for ALL biosamples, in a less subjective slot than the environmental triads, which are from soil or whatever.

**Reply by mslarae13** [reply]
>sample_type says "This can vary from 'environmental package' if the sample is an extraction." which I read as "these two slots are the same unless the sample is an extraction"

This might be outdated since extractions are processed ssamples now.

>I know env_package comes from MIxS. It seems like our [SampleTypeEnum](https://microbiomedata.github.io/nmdc-schema/SampleTypeEnum/) currently captures some MIxS extensions, and our [env_package](https://microbiomedata.github.io/nmdc-schema/env_package/) is currently just a string. Seems like they should both be enums maybe?

agreed

>I'm not fussed about which slot ends up being required. I just think at least one of them should be, so that it's easy to know for ALL biosamples, in a less subjective slot than the environmental triads, which are from soil or whatever.

I advocate for making the env_package (should be extension) required.

**Reply by bmeluch** [reply]
https://github.com/microbiomedata/nmdc-schema/issues/2167

**Comment by bmeluch** [comment]
Issue re: name reconciliation in the submission portal realm https://github.com/microbiomedata/submission-schema/issues/408

**Comment by bmeluch** [comment]
Catching up and re summarizing, since it's been awhile: 

* `nmdc-schema`
    * `sample_type` is recommended and has a range of `SampleTypeEnum`
    * `env_package` is not recommended or required and has a range of `TextValue`
    * `env_package` is deprecated in MIxS
    * most relevant issue https://github.com/microbiomedata/nmdc-schema/issues/2167
    * `sample_type` has been proposed for removal from `nmdc-schema` to only live in `submission-schema` https://github.com/microbiomedata/nmdc-schema/issues/2682 (I think it's meant to represent EMSL types in which case it doesn't 1:1 match the MIxS package list that I think we want to turn into an enum)
* `submission-schema`
    * "The submission portal's frontend code already maintains a crazy mapping between human readable names and `env_package` values and `submission-schema` class names and more. If we could reduce (or at least not increase) the level of that craziness that would be nice." - pkalita on the linked issue above
    * Table from Mark's issue https://github.com/microbiomedata/submission-schema/issues/408
    
| sampleData slot | Interface class | title | excel_worksheet_name | packageName (portal API) | MIxS extension name |
|---|---|---|---|---|---|
| `soil_data` | `SoilInterface` | Soil | soil | soil | soil |
| `water_data` | `WaterInterface` | Water | water | water | water |
| `air_data` | `AirInterface` | Air | air | air | air |
| `biofilm_data` | `BiofilmInterface` | Biofilm | microbial mat_biofilm | ? | microbial mat/biofilm |
| `built_env_data` | `BuiltEnvInterface` | Built Env | built environment | ? | built environment |
| `host_associated_data` | `HostAssociatedInterface` | Host Associated | host-associated | ? | host-associated |
| `misc_envs_data` | `MiscEnvsInterface` | Misc Envs | misc natural or artificial | ? | miscellaneous natural or artificial environment |
| `plant_associated_data` | `PlantAssociatedInterface` | Plant Associated | plant-associated | ? | plant-associated |
| `sediment_data` | `SedimentInterface` | Sediment | sediment | ? | sediment |
| `wastewater_sludge_data` | `WastewaterSludgeInterface` | Wastewater Sludge | wastewater sludge | ? | wastewater/sludge |


Questions/proposals:
* what to do about `env_package` no longer existing in MIxS?
* are we moving `sample_type` to submission-schema only?
* do we need to clean up all of the differences in the table? if so, how? (structured aliases) punctuation/capitalization standardized?
* does this touch on isolate modelling at all?

**Comment by turbomam** [comment]
Refreshing the population numbers, since the ones on slide 3 of the `sample_type_26feb2025` deck are about six months old and two things have changed. Counted today (2026-08-12) against `https://api.microbiomedata.org/nmdcschema/biosample_set`, which holds 27,352 biosamples.

| Slot | Feb 2026 deck | Today |
|---|---|---|
| `samp_type` | None | 0 |
| `sample_type` | None | 456, every one `soil` |
| `env_package` | 2,292 populated, 6,736 NA | 23,069 populated (84.3%) |

Current `env_package` distribution, all 7 distinct values:

```
10689  Metagenome.environmental.1.0
10350  soil
 1292  water
  293  miscellaneous natural or artificial environment
  284  plant-associated
  100  ENVO:00001998
   61  Host-associated
```

Two changes since February:

1. The largest single value is now `Metagenome.environmental.1.0` at 10,689, which is an NCBI package name rather than a MIxS extension name. It arrived with MicroFlora Danica (`nmdc:sty-11-2c3s5473`) and does not appear in the deck's distribution at all. Counting it together with `ENVO:00001998`, 10,789 of the 23,069 populated values (46.8%) are not extension names. `Host-associated` does not match the MIxS extension name `host-associated` either; the deck records that value in lowercase.
2. `sample_type` went from zero records to 456, all of them `soil`. The other four `SampleTypeEnum` values (`soil - water extract`, `plant associated`, `sediment`, `water`) are still unused, and `samp_type` is still zero.

One more datum for whichever slot ends up required: only 5 of the 12 environmental templates the portal offers ever appear in `env_package`. Production has no air, biofilm, built environment, sediment or hydrocarbon values.

Any of these can be re-counted with, for example:

```
curl -sG https://api.microbiomedata.org/nmdcschema/biosample_set \
  --data-urlencode 'filter={"samp_type":{"$exists":true}}' \
  --data-urlencode 'max_page_size=1'
```
